corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +147,117 @@ def data_dictionary_json(request, domain, case_type_name=None):
     })
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_v2(request, domain, case_type_name=None):
+    fhir_resource_type_name_by_case_type = {}
+    fhir_resource_prop_by_case_prop = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type, fhir_resource_prop_by_case_prop = load_fhir_resource_mappings(domain)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
```

Comment:
I assume we only need this when loading data for a specific case type? If so, this can be moved to the conditional where we retrieve the data for a case type and then just return `None` if retrieving the list of case types.

Reply:
Good idea. We could probably return only with case properties I think.
I looked at the code and it seems this refactor will require front end changes and at this point, I am refraining from doing in this PR seeing other JS changes already being done,
Hope that is okay

Reply:
Yeah if this will require too much extensive changing then I agree it's fine to leave as is. Thanks for taking a look.